### PR TITLE
Compositional tidy up

### PR DESCRIPTION
**Pull request by dkachuma**
Tidies up the compositional work.
* Creates a `MultiFluidConstants` space to keep the constants used in fluid calculations in one place.
* This will have to be tied in (possibly removed) with `PhysicsConstants.hpp` from PR #2552

### CODE REVIEW

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CompositionalProperties.hpp [hunk 1]
```diff
@@ -125,6 +121,9 @@ struct CompositionalProperties
                                   real64 & dMassDensity_dp,
                                   real64 & dMassDensity_dt,
                                   arraySlice1d< real64 > const & dMassDensity_dz );
+
+private:
+  static constexpr real64 gasConstant = MultiFluidConstants::gasConstant;
```
Comment (jafranc) [inline src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CompositionalProperties.hpp:126]: As it allows refactoring and you've to pay the price of `#include` anyway, why not bypassing unscoped `gasConstant` and replace its occurences by `MultiFluidConstants::gasConstant`. That would be easier to maintain.

File: src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp [hunk 2]
```diff
@@ -67,9 +68,9 @@ struct CubicEOSPhaseModel
 public:
 
   /// Max number of components allowed in the class for now
-  static constexpr integer maxNumComps = 5;
+  static constexpr integer maxNumComps = MultiFluidConstants::MAX_NUM_COMPONENTS;
   /// Constant for PI
-  static constexpr real64 pi = 3.141592653589793238;
+  static constexpr real64 pi = MultiFluidConstants::pi;
```
Comment (jafranc) [inline src/coreComponents/constitutive/fluid/multifluid/compositional/functions/CubicEOSPhaseModel.hpp:73]: Same note as above

**Review by jafranc — APPROVED** [review]
I would opt for total disparition of class `constexpr` variable for direct use of the single location `static constexpr`. Other than that great and needed PR. Thank you @dkachuma

File: src/coreComponents/constitutive/fluid/multifluid/blackOil/BlackOilFluid.hpp [hunk 3]
```diff
@@ -33,7 +34,7 @@ class BlackOilFluid : public BlackOilFluidBase
 {
 public:
 
-  static constexpr real64 minForPhasePresence = 1e-10;
+  static constexpr real64 minForPhasePresence = MultiFluidConstants::minForSpeciesPresence;
```
Comment (untereiner) [inline src/coreComponents/constitutive/fluid/multifluid/blackOil/BlackOilFluid.hpp:37]: not sure this shortcut needs to be kept. not sure if it is necessary to have the same litteral value stored in the binary 2 times.

File: src/coreComponents/constitutive/fluid/multifluid/PVTDriver.hpp [hunk 4]
```diff
@@ -99,7 +100,8 @@ class PVTDriver : public TaskBase
 
   enum columnKeys { TIME, PRES, TEMP }; ///< Enumeration of "input" column keys for readability
 
-  static constexpr real64 m_baselineTol = 1e-3; ///< Comparison tolerance for baseline results
+  static constexpr real64 m_baselineTol = constitutive::MultiFluidConstants::baselineTolerance; ///< Comparison tolerance for baseline
```
Comment (untereiner) [inline src/coreComponents/constitutive/fluid/multifluid/PVTDriver.hpp:103]: not sure this shortcut needs to be kept. not sure if it is necessary to have the same litteral value stored in the binary 2 times.

File: src/coreComponents/constitutive/fluid/multifluid/MultiFluidBase.hpp [hunk 5]
```diff
@@ -46,14 +47,14 @@ class MultiFluidBase : public ConstitutiveBase
    *
    * @note This puts an upper bound on memory use, allowing to optimize code better
    */
-  static constexpr integer MAX_NUM_COMPONENTS = 16;
+  static constexpr integer MAX_NUM_COMPONENTS = MultiFluidConstants::MAX_NUM_COMPONENTS;
```
Comment (untereiner) [inline src/coreComponents/constitutive/fluid/multifluid/MultiFluidBase.hpp:50]: not sure this shortcut needs to be kept. not sure if it is necessary to have the same litteral value stored in the binary 2 times.

**Review by untereiner — APPROVED** [review]
very much welcome ! I have only a minor feedback.

**Pull request merged**